addons/html_editor/static/src/main/media/icon_plugin.js

```diff
@@ -0,0 +1,118 @@
+                        {
+                            id: "icon_zoom_1",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "1");
+                            },
+                            text: "1x",
+                            name: _t("Icon size 1x"),
+                            isFormatApplied: () => p.hasIconZoom("1"),
+                        },
+                        {
+                            id: "icon_zoom_2",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "2");
+                            },
+                            text: "2x",
+                            name: _t("Icon size 2x"),
+                            isFormatApplied: () => p.hasIconZoom("2"),
+                        },
+                        {
+                            id: "icon_zoom_3",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "3");
+                            },
+                            text: "3x",
+                            name: _t("Icon size 3x"),
+                            isFormatApplied: () => p.hasIconZoom("3"),
+                        },
+                        {
+                            id: "icon_zoom_4",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "4");
+                            },
+                            text: "4x",
+                            name: _t("Icon size 4x"),
+                            isFormatApplied: () => p.hasIconZoom("4"),
+                        },
+                        {
+                            id: "icon_zoom_5",
+                            action(dispatch) {
+                                dispatch("icon_zoom", "5");
+                            },
+                            text: "5x",
+                            name: _t("Icon size 5x"),
+                            isFormatApplied: () => p.hasIconZoom("5"),
+                        },
+                    ],
+                },
+            ],
+        };
+    }
+
+    getSelectedIcon() {
+        const selectedNodes = this.shared.getSelectedNodes();
+        return selectedNodes.find((node) => node.classList.contains("fa"));
+    }
+
+    handleCommand(command, payload) {
+        switch (command) {
+            case "icon_zoom":
+                {
+                    const selectedIcon = this.getSelectedIcon();
+                    if (!selectedIcon) {
+                        return;
+                    }
+                    for (const classString of selectedIcon.classList) {
+                        if (classString.match(/^fa-[0-9]x$/)) {
+                            selectedIcon.classList.remove(classString);
+                        }
+                    }
+                    if (payload !== "1") {
+                        selectedIcon.classList.toggle(`fa-${payload}x`);
```

Comment:
if I got this right, `toggle` here can only mean `add` (and not "`remove` if present, `add` otherwise", as you removed in the loop above). So I would suggest using `add` instead for better clarity.